### PR TITLE
Table data in redux

### LINKED ISSUE #2103: Add Redux to experiments table
Now that the plots state reside in Redux, we should do the same with the experiments table so that we can work on both webviews in the same way.

### PR DESCRIPTION
**Pull request by sroy3**
Closes #2103

### CODE REVIEW

File: webview/src/experiments/components/table/tableDataSlice.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  initialState: tableDataInitialState,
+  name: 'tableData',
+  reducers: {
+    update: (state, action: PayloadAction<TableData>) => {
+      if (action.payload) {
+        return {
+          ...state,
+          ...action.payload,
+          hasData: true
+        }
```
Comment (sroy3) [inline webview/src/experiments/components/table/tableDataSlice.ts:36]: This is pretty much the same as it was before. An improvement here would be to go through previous columns, rows and objects to change the reference of only what changed. I will do that as a follow up.

**Review by julieg18 — APPROVED** [review]
Great work! Things look more readable with less props being passed around :)

**Review by mattseddon — APPROVED** [review]
[Q] How does performance compare? Looks good to go otherwise.

**Comment by sroy3** [comment]
> [Q] How does performance compare? Looks good to go otherwise.

I'm not seeing any difference. It's pretty much just a drop in replacement since the whole thing changes when the data changes. Like I said I'll follow up to try to limit the changes to only what changes. That could help with really big project, but that is not something we will notice otherwise.

**Pull request merged**